libraries/ts-command-line/src/providers/CommandLineParameterProvider.ts

```diff
@@ -456,6 +486,72 @@ export abstract class CommandLineParameterProvider {
+      if (data[parserKey]) {
+        // Determine if the ambiguous parameter is a short name or a long name, since the process of finding
+        // the non-ambiguous name is different for each.
+        const duplicateShortNameParameters: CommandLineParameter[] | undefined =
+          this._parametersByShortName.get(parameterName);
+        if (duplicateShortNameParameters) {
+          // We also need to make sure we get the non-ambiguous long name for the parameter, since it is
+          // possible for that the long name is ambiguous as well.
+          const nonAmbiguousLongNames: string[] = [];
+          for (const parameter of duplicateShortNameParameters) {
+            const matchingLongNameParameters: CommandLineParameter[] | undefined =
+              this._parametersByLongName.get(parameter.longName);
+            if (!matchingLongNameParameters?.length) {
+              // This should never happen
+              throw new Error(
+                `Unable to find long name parameters for ambiguous short name parameter "${parameterName}".`
+              );
+            }
+            // If there is more than one matching long name parameter, then we know that we need to use the
+            // scoped long name for the parameter. The scoped long name should always be provided.
+            if (matchingLongNameParameters.length > 1) {
+              if (!parameter.scopedLongName) {
+                // This should never happen
+                throw new Error(
+                  `Unable to find scoped long name for ambiguous short name parameter "${parameterName}".`
+                );
+              }
+              nonAmbiguousLongNames.push(parameter.scopedLongName);
+            } else {
+              nonAmbiguousLongNames.push(parameter.longName);
+            }
+          }
+          throw new Error(
+            `The provided parameter "${parameterName}" is ambiguous. It could reference any of ` +
+              `the following parameters: ${nonAmbiguousLongNames.join(', ')}.`
+          );
+        }
+
+        const duplicateLongNameParameters: CommandLineParameter[] | undefined =
+          this._parametersByLongName.get(parameterName);
+        if (duplicateLongNameParameters) {
+          const nonAmbiguousLongNames: string[] = duplicateLongNameParameters.map(
+            (p: CommandLineParameter) => {
+              // The scoped long name should always be provided
+              if (!p.scopedLongName) {
+                // This should never happen
+                throw new Error(
+                  `Unable to find scoped long name for ambiguous long name parameter "${parameterName}".`
+                );
+              }
+              return p.scopedLongName;
+            }
+          );
+          throw new Error(
+            `The provided parameter "${parameterName}" is ambiguous. It could reference any of ` +
+              `the following parameters: ${nonAmbiguousLongNames.join(', ')}.`
```

Comment:
Suggested rewording:
```
The parameter name "--do-thing" is ambiguous. Instead use one of
these scoped names: "--jest:do-thing", "--storybook:do-thing"
```

Note that  `nonAmbiguousLongNames.join(', ')` needs to include quotes for consistentency